src/purchase-selection/utils.ts

```diff
@@ -0,0 +1,155 @@
+import {PreassignedFareProduct} from '@atb/configuration/types';
+import {compareVersion} from '@atb/utils/compare-version';
+import type {
+  PurchaseSelectionBuilderInput,
+  PurchaseSelectionType,
+} from './types';
+import {TariffZoneWithMetadata} from '@atb/tariff-zones-selector';
+import turfBooleanPointInPolygon from '@turf/boolean-point-in-polygon';
+import {UserProfileWithCount} from '@atb/fare-contracts';
+import {TariffZone, UserProfile} from '@atb-as/config-specs';
+import {isValidDateString} from '@atb/utils/date';
+
+export const isProductSellableInApp = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+) => {
+  if (
+    (product.limitations.appVersionMin &&
+      compareVersion(product.limitations.appVersionMin, input.appVersion) >
+        0) ||
+    (product.limitations.appVersionMax &&
+      compareVersion(product.limitations.appVersionMax, input.appVersion) < 0)
+  )
+    return false;
+
+  if (
+    product.distributionChannel.some((channel) => channel === 'debug-app') &&
+    input.customerProfile?.debug
+  )
+    return true;
+
+  return product.distributionChannel.some((channel) => channel === 'app');
+};
+
+export const getDefaultProduct = (
+  input: PurchaseSelectionBuilderInput,
+  configType: string,
+) =>
+  input.preassignedFareProducts
+    .filter((product) => isProductSellableInApp(input, product))
+    .filter((product) => product.type === configType)
+    .reduce((selected, current) => (current.isDefault ? current : selected));
+
+export const getDefaultZone = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+): TariffZoneWithMetadata => {
+  const selectableZones = input.tariffZones.filter((zone) =>
+    isSelectableZone(product, zone),
+  );
+
+  if (input.currentCoordinates) {
+    const {longitude, latitude} = input.currentCoordinates;
+    const zoneFromLocation = selectableZones.find((t) =>
+      turfBooleanPointInPolygon([longitude, latitude], t.geometry),
+    );
+    if (zoneFromLocation) {
+      return {...zoneFromLocation, resultType: 'geolocation'};
+    }
+  }
+
+  const zone = selectableZones.reduce((selected, current) =>
+    current.isDefault ? current : selected,
+  );
+  return {...zone, resultType: 'zone'};
+};
+
+export const getDefaultUserProfiles = (
+  input: PurchaseSelectionBuilderInput,
+  product: PreassignedFareProduct,
+): UserProfileWithCount[] => {
+  const selectableProfiles = getSelectableUserProfiles(
+    input.userProfiles,
+    product,
+  );
+  const profile = selectableProfiles.reduce((selected, current) =>
+    current.userTypeString === input.defaultUserTypeString ? current : selected,
+  );
+  return [{...profile, count: 1}];
```

Comment:
If only returns one profile why it expect an array?

Reply:
Yeah, this helper could have been `getDefaultUserProfile` and only returned a single profile, and then builder it self could map it to `[{...profile, count: 1}]`. But this is a helper to make the builder as slim/clean as possible, and from the viewpoint of the builder it needs default value for the field `userProfiles`, and then it makes sense to have it as a `getDefaultUserProfiles` function.